File: Adventure.Persistence/Containers/Item.cs
using Microsoft.Azure.Documents;
using Newtonsoft.Json;

namespace Adventure.Persistence.Containers
{
    public class Item : Resource
    {
        [JsonProperty(PropertyName = "parentId")]
        public string ParenId { get; set; }

        [JsonProperty(PropertyName = "value")]
        public string Value { get; set; }

        [JsonProperty(PropertyName = "children")]
        public Children[] Children { get; set; }
    }

    public class Children
    {
        [JsonProperty(PropertyName = "id")]
        public string Id { get; set; }

        [JsonProperty(PropertyName = "value")]
        public string Value { get; set; }
    }

    public class Doughnut : Item
    { }
}

File: AdventureApp/my-own-adventure-app/src/app/adventure-details/item.ts
export interface IItem {
    id: string;
    parentId: string;
    value: string;
    cssClass: string;
    children: IItem[];
  }

  export class Item implements IItem {
    id: string;
    parentId: string;
    value: string;
    cssClass: string;
    children: IItem[];
  }